src/main/java/com/direwolf20/buildinggadgets/common/tools/InventoryManipulation.java

```diff
@@ -130,11 +132,16 @@ public static boolean useItem(ItemStack target, EntityPlayer player, int amountR
         return amountLeft < amountRequired;
     }
 
-    private static int extractFromInventory(IItemHandler inventory, ItemStack target, int amountRequired) {
+    private static int extractFromInventory(IItemHandler inventory, ItemStack target, int amountRequired, EntityPlayer player) {
         int amountSaturated = 0;
         if( inventory == null )
             return amountSaturated;
 
+        if(inventory instanceof IBlockProviderItemHandler) {
```

Comment:
Instead of short-curcuiting here for an IBlockProviderItemHandler, could you wrap non-IBlockProviderItemHandlers in a delegating adapter implementation and directly use it's Method? My suggestion for the default Method in the interface wouldn't require more then an implementation which delegates all of the regular IITemHandler Methods.

Reply:
I mean, sure. But I think it would increase code complexity by a lot and not make it more readable necessarily. Yes, it would make the code shorter overall, but I doubt it would make it more readable.

Reply:
My thoughts here extactly. I'd rather these special case inventories aren't dumped into the same logic as normal and typical inventories.

Reply:
You in this case, although I see the code advantages, I don't see a reason to go all over the top on a Limited Support version of the mod, as it is, I am only really wanting to work on 1.12 because of how much I like the Builder's bag mod :D and a few bugs I really want to fix.

Reply:
I have no idea what you're trying to tell us here... sorry

Reply:
Essentially saying it's not worth the time nor effort to do that. I may modify things in the future but for now, na.